### PR TITLE
[SYCLomatic] Optimize the for loop migration by adding #pragma unroll

### PR DESCRIPTION
**Pull request by intwanghao**

### CODE REVIEW

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -14744,3 +14744,58 @@ void CudaUuidRule::runRule(
...
 REGISTER_RULE(CudaUuidRule, PassKind::PK_Analysis)
+
+void ForLoopUnrollRule::registerMatcher(ast_matchers::MatchFinder &MF) {
+  MF.addMatcher(forStmt().bind("loop"), this);
+}
+
+void ForLoopUnrollRule::runRule(
+    const ast_matchers::MatchFinder::MatchResult &Result) {
+  if (!DpctGlobalInfo::isOptimizeMigration()) {
+    return;
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ASTTraversal.cpp]: can move the check into like L14749.   Also consider to add the rule into separate file.

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 2]
```diff
@@ -14744,3 +14744,58 @@ void CudaUuidRule::runRule(
 }
 
 REGISTER_RULE(CudaUuidRule, PassKind::PK_Analysis)
+
+void ForLoopUnrollRule::registerMatcher(ast_matchers::MatchFinder &MF) {
+  MF.addMatcher(forStmt().bind("loop"), this);
+}
+
+void ForLoopUnrollRule::runRule(
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ASTTraversal.cpp]: please add comment on the high level algorithm to insert the #pragma unroll.

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 3]
```diff
@@ -14744,3 +14744,58 @@ void CudaUuidRule::runRule(
...
+    auto FD = getImmediateOuterFuncDecl(Loop);
+    if (!FD ||
+        (!FD->hasAttr<CUDAGlobalAttr>() && !FD->hasAttr<CUDADeviceAttr>())) {
+      return;
+    }
+
+    if (auto Parent = Context.getParents(*Loop)[0].get<AttributedStmt>()) {
+      auto Attrs = Parent->getAttrs();
+      for (auto Attr : Attrs) {
+        if (dyn_cast<LoopHintAttr>(Attr)) {
```
Comment (ziranzha) [inline clang/lib/DPCT/ASTTraversal.cpp]:
```suggestion
        if (isa<LoopHintAttr>(Attr)) {
```

**Review by ziranzha — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**